src/Marten.Testing/Harness/DestructiveIntegrationContext.cs

```diff
@@ -0,0 +1,23 @@
+using Xunit;
+
+namespace Marten.Testing.Harness
+{
+    /// <summary>
+    /// Use this base class if a test fixture needs to do anything
+    /// destructive to the default database schema
+    /// </summary>
+    [Collection("integration")]
+    public class DestructiveIntegrationContext: IntegrationContext
+    {
+        public DestructiveIntegrationContext(DefaultStoreFixture fixture) : base(fixture)
+        {
+            theStore.Advanced.Clean.CompletelyRemoveAll();
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
+            theStore.Advanced.Clean.CompletelyRemoveAll();
```

Comment:
**CONCERN:** Do we need to call this also on dispose if we're running it also in constructor? That seems to be potentially redundant.

Reply:
We don't do the remove all in the constructor function, and this only applies if someone called `StoreOptions()` that could quite possibly make destructive changes to the public schema. This was a workaround for IntegrationContext sometimes being used for altered configuration in the same class.